### ISSUE TITLE
React compatibility

### ISSUE THREAD

**Issue by sasirura** [antoine92190/vue-advanced-chat#233]
## Describe the bug

I imported the plugin to react as a web component. 

```
import React, { useEffect, useRef, useState } from "react";
import "../../../node_modules/vue-advanced-chat/dist/vue-advanced-chat.min";
import "./chat.css";

function Test() {
    const [rooms, setRooms] = useState([
        {
            roomId: 1,
            roomName: "Room 1",
            avatar: "assets/imgs/people.png",
            unreadCount: 4,
            index: 3,
            users: [
                {
                    _id: 1234,
                    username: "John Doe",
                    avatar: "assets/imgs/doe.png",
                    status: {
                        state: "online",
                        lastChanged: "today, 14:30"
                    }
                },
                {
                    _id: 4321,
                    username: "John Snow",
                    avatar: "assets/imgs/snow.png",
                    status: {
                        state: "offline",
                        lastChanged: "14 July, 20:00"
                    }
                }
            ],
        }
    ]);
    
    const [messages, setMessages] = useState([{}]);

    function invokeCallback({ detail }) {
        console.log(detail);
    }
      
    // hook to maintain chat component instance
    const chatElm = useRef(null);
    
    useEffect(() => {
        if (chatElm.current) {
            // Update the rooms property of the chat web component imperatively
            chatElm.current.rooms = rooms;
            chatElm.current.messages = messages;
            chatElm.current.styles = styles;
            chatElm.current.addEventListener("send-message", invokeCallback);
            
            
        }
    }, [rooms, messages]);

  
    
    return (
        <div className ="chat">
            <vue-advanced-chat 
                current-user-id= { 4321 }
                ref ={ chatElm }
                show-send-icon = { true }
                show-emojis = { true }
            />
        </div>
    );
}
export default Test;
```
 
After that I can not click the send icon. it is not enabled

![image](https://user-images.githubusercontent.com/42542164/139024775-500e2318-00d0-46ad-986e-df46913631e2.png)

**Comment by antoine92190** [comment]
There is probably a compatibility issue with react, I will work on it and see if I can fix it.

**Comment by dahromy** [comment]
I have the same error in nuxtjs

**Comment by KeeganFargher** [comment]
@sasirura Did you ever manage to find a workaround for this? Experiencing the same issue. Weirdly, if I attach an image the send button becomes enabled.

**Comment by KeeganFargher** [comment]
Looks like on this line

```
onChangeInput: debounce(function (e) {
	if (e?.target?.value || e?.target?.value === '') {
		this.message = e.target.value
	}
	this.keepKeyboardOpen = true
	this.resizeTextarea()
	this.$emit('typing-message', this.message)
}, 100),
```

`e.target.value` returns `vue-advanced-chat`, not the textarea input, which I guess is why the value is always undefined?

**Comment by KeeganFargher** [comment]
update...

I changed the onChangeInput code to:

```
onChangeInput: debounce(function (e) {
	if (this.getTextareaRef()?.value || this.getTextareaRef()?.value === '') {
		this.message = this.getTextareaRef()?.value
	}
	this.keepKeyboardOpen = true
	this.resizeTextarea()
	this.$emit('typing-message', this.message)
}, 100),
```
And it's working now.

**Comment by antoine92190** [comment]
@KeeganFargher Thanks, it seems to do the trick, I fixed it in https://github.com/antoine92190/vue-advanced-chat/commit/f0420fb261b910171e1098ddaaf63956df070ddf

**Comment by sasirura** [comment]
@KeeganFargher  @antoine92190  Thank you guys !

**Comment by rajeev-ayer** [comment]
This is a great plugin.  Thank you very much for your contributions.

However, I am experiencing the exact same issue with Angular.  The send button is disabled after typing text except when I attach a file.

I downloaded the package just yesterday (29 Nov 2021) so I am assuming I have the latest version (1.4.2) which includes this fix for React.  Wondering if there is another issue with Angular.

In addition, I get an error when clicking on the emoji icon next to the textarea.  I get a "Cannot read properties of null (reading 'addEventListener')".  

I figured I would post this here as this seems to be a compatibility issue as well.  However, if it should be moved to a separate issue please do so or let me know and I can open another issue.

**Comment by KeeganFargher** [comment]
@rajeev-ayer It's because a lot of the methods call `document.getElementById` but since it's a web component it's inside a shadow dom, and while in the shadow dom you cannot access `document`.

As a really hacky solution, I replaced most `document.getElementById` with `document.querySelector('vue-advanced-chat').shadowRoot.getElementById`, which has been working but I haven't tested it much.

I don't really know much about web components and shadow dom so someone correct me if I'm wrong here?

**Comment by rajeev-ayer** [comment]
I believe you are correct @KeeganFargher.

I cloned the repository and modified the files locally to how to best get around this issue.  You were on the right track, but instead of using `document.querySelector(...).shadowRoot`, it works if you use the $refs vue property instead; e.g. `this.$refs.emojiPicker.shadowRoot` which gives you access to the DOM object.

In the 

I am currently working on Angular project and would like to continue using this component.  Therefore, I am happy to contribute to this project and fix any issues that I might encounter when the component is used within an Angular project.  If you are all amenable to that, please let me know.

**Comment by antoine92190** [comment]
@rajeev-ayer You are more than welcome to send a PR with compatibility fixes for Angular. Unfortunately I don't have time to test the component on Vue, React & Angular for every release, so I'm happy for all the help I can get :)

**Comment by antoine92190** [comment]
In that case, please open an issue "Angular compatibility" so we can move the conversation in a more relevant channel.

**Comment by rajeev-ayer** [comment]
Agreed @antoine92190.  I have opened issue #260 to cover this problem.

> @rajeev-ayer You are more than welcome to send a PR with compatibility fixes for Angular. Unfortunately I don't have time to test the component on Vue, React & Angular for every release, so I'm happy for all the help I can get :)

I will submit a PR accordingly with the fix for issue #260.
